worlds/stardew_valley/content/mods/archeology.py

```diff
@@ -1,14 +1,43 @@
-from ..game_content import ContentPack
+from ..game_content import ContentPack, StardewContent
 from ..mod_registry import register_mod_content_pack
+from ...data.artisan import MachineSource
 from ...data.game_item import ItemTag, Tag
 from ...data.shop import ShopSource
 from ...data.skill import Skill
 from ...mods.mod_data import ModNames
 from ...strings.book_names import ModBook
+from ...strings.craftable_names import ModCraftable, ModMachine
+from ...strings.fish_names import ModTrash
+from ...strings.metal_names import all_artifacts, all_fossils
 from ...strings.region_names import LogicRegion
 from ...strings.skill_names import ModSkill
 
-register_mod_content_pack(ContentPack(
+
+class ArchaeologyContentPack(ContentPack):
+    def artisan_good_hook(self, content: StardewContent):
+        # Done as honestly there are too many display items to put into the initial registration traditionally.
+        display_types = [ModCraftable.wooden_display, ModCraftable.hardwood_display]
+        display_items = all_artifacts + all_fossils
+        rusty_scrap_machine_rules = []
+        for item in display_items:
+            if item in all_artifacts:
```

Comment:
Since `all_artifacts` is a list, to find whether an item is part of it or not, behind the scene it will iterate through the whole list each time, which is not really fast. 

It would be better to move it to its own for loop on the artifact directly, so we don't have to check if the item is an artifact or not. It could also be done with a generator in the same line when adding the sources with something like: 
```python
content.source_item(ModTrash.rusty_scrap, *(MachineSource(item=artifact, machine=ModMachine.grinder) for artifact in all_artifacts))
```

Reply:
Believe I've done this.  Also split off the iteration and made it a bit different.  I won't resolve this yet so you know it can be looked over again.